### PR TITLE
Update alias parsing and add runbook to create a SKU-less bundle.

### PR DESCRIPTION
**Pull request by TheSchnoo**
## Description
- Updates alias parsing to support extracting IDs from existing resources for nested component identification
- Adds discussion on this alias parsing to `tutorial.md`
- Adds the `publish-catalog-with-bundles` runbook action to automate and simplify bundle creation
- Updates the `reset` runbook action to remove entities created in `publish-catalog-with-bundles`

## Testing
- New and existing unit tests
- Tested against deployed environment to verify expected resource creation/deletion

### CODE REVIEW

File: external/resources/resources.yaml [hunk 1]
```diff
@@ -1553,6 +1553,8 @@ pcm-products:
       type: STRING
     locales.en-US.description:
       type: STRING
+    components.*.options[n].id:
+      type: RESOURCE_ID:pcm-products
```
Comment (steve-r-west) [inline external/resources/resources.yaml:1557]:
```suggestion
```

Let's drop this for now, and create a distinct bug.

File: external/runbooks/pxm-how-to.epcc.yml [hunk 2]
```diff
@@ -84,6 +84,26 @@ actions:
...
+        epcc create pcm-product-price name=VIP_Pricing currencies.USD.amount 500 currencies.USD.includes_tax false sku squeaky-ball-sku
+      - epcc create pcm-product attributes.name "SKU-less Bed and Ball Bundle" attributes.commodity_type "physical" status "live" components.dogbed.max 1 components.dogbed.min 1 components.dogbed.name "Dog Bed" components.dogbed.options[0].id alias/product/name=Fluffy_Bed/id components.dogbed.options[0].type "product" components.dogbed.options[0].quantity 1 components.dogball.max 1 components.dogball.min 1 components.dogball.name "Dog Ball" components.dogball.options[0].id alias/product/name=Squeaky_Ball/id components.dogball.options[0].type "product" components.dogball.options[0].quantity 1
+      - epcc create pcm-node-product name=Pet_Supplies name=Dog_Supplies data[0].type product data[0].id name=SKU-less_Bed_and_Ball_Bundle
+      - epcc create pcm-catalog name "Pet Supplies Catalog" description "Pet Supplies Catalog" pricebook_id name=VIP_Pricing hierarchy_ids[0] name=Pet_Supplies
+      - epcc create pcm-catalog-releases name=Pet_Supplies_Catalog
+      - epcc get pcm-catalog-releases name=Pet_Supplies_Catalog
```
Comment (steve-r-west) [inline external/runbooks/pxm-how-to.epcc.yml:105]:
```suggestion
      - sleep 2
      - epcc get pcm-catalog-releases name=Pet_Supplies_Catalog
```

Just give PCM a chance to publish the catalog.

**Review by steve-r-west — APPROVED** [review]

File: external/resources/resources.yaml (hunk 1 above)
Comment (TheSchnoo) [reply external/resources/resources.yaml:1557]: I might have misunderstood during our discussion, but won't removing this prevent the new runbook action from working correctly?

File: external/runbooks/pxm-how-to.epcc.yml (hunk 2 above)
Comment (TheSchnoo) [reply external/runbooks/pxm-how-to.epcc.yml:106]: Added, thanks!

File: external/resources/resources.yaml (hunk 1 above)
Comment (steve-r-west) [reply external/resources/resources.yaml:1557]: > I might have misunderstood during our discussion

Yes and that is 100% my fault, I was guiding you towards one solution and then telling you a whole bunch of irrelevant details, and then we switched solutions. We should create a new bug for the above, (and I think you did already). 

The above is needed but we need to support some things with regular expressions, and auto complete for it. You don't need it to submit the JSON, it's just used for auto complete, and a few other things.

**Pull request merged**